class.jetpack.php

```diff
@@ -1302,8 +1302,8 @@ public static function is_development_mode() {
 			$development_mode = JETPACK_DEV_DEBUG;
 		}
 
-		elseif ( site_url() && false === strpos( site_url(), '.' ) ) {
-			$development_mode = true;
+		elseif ( $site_url = site_url() ) {
```

Comment:
I don't know—it's original to the function as committed to Jetpack https://github.com/Automattic/jetpack/commit/eff10c9791769b41edc17288f850d5739543e076